guardrails/schema/json_schema.py

```diff
@@ -174,13 +194,30 @@ def from_xml(
     @classmethod
     def from_pydantic(
         cls,
-        model: Type[BaseModel],
+        model: Union[Type[BaseModel], Type[List[Type[BaseModel]]]],
         reask_prompt_template: Optional[str] = None,
         reask_instructions_template: Optional[str] = None,
     ) -> Self:
         strict = False
 
-        schema = convert_pydantic_model_to_datatype(model, strict=strict)
+        type_origin = get_origin(model)
+
+        if type_origin == list:
+            item_types = get_args(model)
+            if len(item_types) > 1:
+                raise ValueError("List data type must have exactly one child.")
```

Comment:
If I understand this correctly, we're defining lists as only being able to support a singular type - is that the correct understanding? If so, why do we need to enforce that?

Reply:
correct - lists support a singular type. we enforce it so that users expect the correct behavior. if they pass in multiple types, anything after the first type will be ignored.